src/OmnichannelChatSDK.ts

```diff
@@ -1693,6 +1635,146 @@ class OmnichannelChatSDK {
         }
     }
 
+    public async getAgentAvailability(optionalParams: GetAgantAvailabilityOptionalParams = {}): Promise<any> { // eslint-disable-line @typescript-eslint/no-explicit-any
+        const requestId = uuidv4();
+        
+        this.scenarioMarker.startScenario(TelemetryEvent.GetAgentAvailability, {
+            RequestId: requestId
+        });
+
+        if (!this.authSettings) {
+            const exceptionDetails: ChatSDKExceptionDetails = {
+                response: "UnAuthUnsupported",
+                message: "GetAgentAvailability is supported only for authenticated live chat widget."
+            };
+
+            this.scenarioMarker.failScenario(TelemetryEvent.GetAgentAvailability, {
+                RequestId: requestId,
+                ExceptionDetails: JSON.stringify(exceptionDetails)
+            });
+
+            throw new Error(exceptionDetails.message);
+        }
+
+        if (!this.authenticatedUserToken) {
+            const exceptionDetails = {
+                response: "UndefinedAuthToken",
+                message: "Missing AuthToken for GetAgentAvailability."
+            }
+
+            this.scenarioMarker.failScenario(TelemetryEvent.GetAgentAvailability, {
+                RequestId: requestId,
+                ExceptionDetails: JSON.stringify(exceptionDetails)
+            });
+
+            throw new Error(exceptionDetails.message);
+        }
+
+        if (this.conversation) {
+            const exceptionDetails = {
+                response: "InvalidOperation",
+                message: "GetAgentAvailability can only be called before a chat has started."
+            }
+
+            this.scenarioMarker.failScenario(TelemetryEvent.GetAgentAvailability, {
+                RequestId: requestId,
+                ChatId: this.chatToken.chatId as string,
+                ExceptionDetails: JSON.stringify(exceptionDetails)
+            });
+
+            throw new Error(exceptionDetails.message);
```

Comment:
this code is repeated 3 times in this function, could we find a way to abstract it?

Reply:
refactored